Apps/Playground/App.tsx

```diff
@@ -1,118 +1,102 @@
 /**
- * Sample React Native App
- * https://github.com/facebook/react-native
- *
  * Generated with the TypeScript template
  * https://github.com/react-native-community/react-native-template-typescript
  *
  * @format
  */
 
-import React from 'react';
-import {
-  SafeAreaView,
-  StyleSheet,
-  ScrollView,
-  View,
-  Text,
-  StatusBar,
-} from 'react-native';
+import React, { useState, FunctionComponent, useEffect } from 'react';
+import { SafeAreaView, StatusBar, Button, View, Text, ViewProps } from 'react-native';
+
+import { EngineView, useEngine } from 'react-native-babylon';
+import { Scene, Vector3, Mesh, ArcRotateCamera, Engine, Camera, PBRMetallicRoughnessMaterial, Color3 } from '@babylonjs/core';
+import Slider from '@react-native-community/slider';
+
+const EngineScreen: FunctionComponent<ViewProps> = (props: ViewProps) => {
+  const [toggleView, setToggleView] = useState(false);
+  const [camera, setCamera] = useState<Camera>();
+  const [box, setBox] = useState<Mesh>();
+  const [scale, setScale] = useState<number>(1);
+  const [fps, setFps] = useState<number>(0);
+
+  useEngine((engine: Engine) => {
+    const scene = new Scene(engine);
+    scene.createDefaultCamera(true);
+    if (scene.activeCamera != null) {
+      (scene.activeCamera as ArcRotateCamera).beta -= Math.PI / 8;
+      setCamera(scene.activeCamera);
+    }
+    scene.createDefaultLight(true);
+
+    const box = Mesh.CreateBox("box", 0.3, scene);
+    setBox(box);
+    const mat = new PBRMetallicRoughnessMaterial("mat", scene);
+    mat.metallic = 1;
+    mat.roughness = 0.5;
+    mat.baseColor = Color3.Red();
+    box.material = mat;
 
-import {
-  Header,
-  LearnMoreLinks,
-  Colors,
-  DebugInstructions,
-  ReloadInstructions,
-} from 'react-native/Libraries/NewAppScreen';
+    scene.beforeRender = function () {
+      scene.meshes[0].rotate(Vector3.Up(), 0.005 * scene.getAnimationRatio());
+    };
 
-declare var global: {HermesInternal: null | {}};
+    const timerHandle = setTimeout(() => {
+      setFps(engine.getFps());
+    }, 1000);
+
+    return () => {
+      clearTimeout(timerHandle);
+    };
```

Comment:
Are you trying to show the FPS regularly? I'm not sure I understand what this is doing.

Reply:
Yes, it was a last minute addition that I messed up. Updated to `setInterval`.